kpi/views/v2/asset.py

```diff
@@ -554,16 +554,22 @@ def finalize_response(self, request, response, *args, **kwargs):
         # model-level permissions, accepted_renderer won't be present.
         if hasattr(request, 'accepted_renderer'):
             # Check the class of the renderer instead of just looking at the
-            # format, because we don't want to set Content-Disposition:
-            # attachment on asset snapshot XML
-            if (isinstance(request.accepted_renderer, XlsRenderer) or
-                    isinstance(request.accepted_renderer, XFormRenderer)):
+            # format, because we don't want to set
+            # `Content-Disposition: attachment` on asset snapshot XML
+            if isinstance(request.accepted_renderer, XFormRenderer):
+                filename = '.'.join(
+                    (self.get_object().uid, request.accepted_renderer.format)
+                )
                 response[
                     'Content-Disposition'
-                ] = 'attachment; filename={}.{}'.format(
-                    self.get_object().uid,
-                    request.accepted_renderer.format
-                )
+                ] = 'attachment; filename={}'.format(filename)
+            if isinstance(request.accepted_renderer, XlsRenderer):
+                # `accepted_renderer.format` is 'xls' here for historical
+                # reasons, but what we actually serve now is xlsx (Excel 2007+)
+                filename = '.'.join((self.get_object().uid, 'xlsx'))
+                response[
+                    'Content-Disposition'
+                ] = 'attachment; filename={}'.format(filename)
```

Comment:
What about `f-string`?

Reply:
I agree that it'd be better, but I see myself having to backport this to Python 2 😐